src/main/java/io/vertx/blueprint/todolist/verticles/TodoVerticle.java

```diff
@@ -0,0 +1,240 @@
+        notFound(context);
+      else {
+        final String encoded = Json.encodePrettily(res.get());
+        context.response()
+          .putHeader("content-type", "application/json; charset=utf-8")
+          .end(encoded);
+      }
+    }));
+  }
+
+  private void handleGetAll(RoutingContext context) {
+    service.getAll().setHandler(resultHandler(context, res -> {
+      if (res == null) {
+        serviceUnavailable(context);
+      } else {
+        final String encoded = Json.encodePrettily(res);
+        context.response()
+          .putHeader("content-type", "application/json; charset=utf-8")
+          .end(encoded);
+      }
+    }));
+  }
+
+  private void handleUpdateTodo(RoutingContext context) {
+    try {
+      String todoID = context.request().getParam("todoId");
+      final Todo newTodo = Utils.getTodoFromJson(context.getBodyAsString());
+      // handle error
+      if (newTodo == null || todoID == null) {
+        sendError(400, context.response());
+        return;
+      }
+      service.update(todoID, newTodo)
+        .setHandler(resultHandler(context, res -> {
+          if (res == null)
+            notFound(context);
+          else {
+            final String encoded = Json.encodePrettily(res);
+            context.response()
+              .putHeader("content-type", "application/json; charset=utf-8")
+              .end(encoded);
+          }
+        }));
+    } catch (DecodeException e) {
+      badRequest(context);
+    }
+  }
+
+  private Handler<AsyncResult<Boolean>> deleteResultHandler(RoutingContext context) {
+    return res -> {
+      if (res.succeeded()) {
+        if (res.result()) {
+          context.response().setStatusCode(204).end();
+        } else {
+          serviceUnavailable(context);
+        }
+      } else {
+        serviceUnavailable(context);
+      }
+    };
+  }
+
+  private void handleDeleteOne(RoutingContext context) {
+    String todoID = context.request().getParam("todoId");
+    service.delete(todoID)
+      .setHandler(deleteResultHandler(context));
+  }
+
+  private void handleDeleteAll(RoutingContext context) {
+    service.deleteAll()
+      .setHandler(deleteResultHandler(context));
+  }
+
+  private void sendError(int statusCode, HttpServerResponse response) {
+    response.setStatusCode(statusCode).end();
+  }
+
+  private void notFound(RoutingContext context) {
+    context.response().setStatusCode(404).end();
+  }
+
+  private void badRequest(RoutingContext context) {
+    context.response().setStatusCode(400).end();
+  }
+
+  private void serviceUnavailable(RoutingContext context) {
+    context.response().setStatusCode(503).end();
+  }
+
+  /**
+   * Wrap the Todo entity with appropriate id and url
+   *
+   * @param todo    a todo entity
+   * @param context RoutingContext
+   * @return the wrapped todo entity
+   */
+  private Todo wrapObject(Todo todo, RoutingContext context) {
+    if (todo.getId() == 0)
+      todo.setId(Math.abs(new Random().nextInt()));
```

Comment:
Well sorry...I thought the UUID as id might not be efficient for some persistence(e.g. MySQL) but I forgot the duplicates of random... What if we use a static `AtomicInteger` as the incremental id counter?